my_dev_t is used only in struct loop_info definition (see <linux/loop.h>).
It has to be _old_ kernel dev_t type (i.e. used before 2.6.x).
For 2.6.x headers use __kernel_old_dev_t (for x86: 16-bit, while
__kernel_dev_t is 32-bit and recent glibc dev_t is 64-bit)

--- util-linux-2.12.orig/mount/my_dev_t.h	2003-07-16 22:05:50.000000000 +0200
+++ util-linux-2.12/mount/my_dev_t.h	2004-12-19 14:31:34.467427960 +0100
@@ -4,4 +4,4 @@
 /* for ancient systems use "unsigned short" */
 
 #include <linux/posix_types.h>
-#define my_dev_t __kernel_dev_t
+#define my_dev_t __kernel_old_dev_t
